internal/wts.go

```diff
@@ -0,0 +1,98 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type Gen3FuseConfig struct {
+	LogFilePath string `yaml:"LogFilePath"`
+
+	// Workspace Token Service configuration
+	WTSBaseURL          string `yaml:"WTSBaseURL"`
+	WTSFenceConnectPath string `yaml:"WTSFenceConnectPath"`
+	WTSAccessTokenPath  string `yaml:"WTSAccessTokenPath"`
+
+	// Fence configuration
+	FenceBaseURL          string `yaml:"FenceBaseURL"`
+	FencePresignedURLPath string `yaml:"FencePresignedURLPath"`
+
+	// Indexd configuration
+	IndexdBaseURL          string `yaml:"IndexdBaseURL"`
+	IndexdBulkFileInfoPath string `yaml:"IndexdBulkFileInfoPath"`
+}
+
+func (gc *Gen3FuseConfig) GetGen3FuseConfigFromYaml(filename string) (err error) {
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil || yamlFile == nil {
+		FuseLog("yamlFile.Get err: " + err.Error())
+		return err
+	}
+
+	err = yaml.Unmarshal(yamlFile, gc)
+	if err != nil {
+		FuseLog("Unmarshal: " + err.Error())
+		return err
+	}
+
+	return nil
+}
+
+type fenceConnectResponse struct {
+	Success string
+	Error   string
+}
+
+type tokenResponse struct {
+	Token string
+}
+
+var myClient = &http.Client{Timeout: 10 * time.Second}
+
+func getJson(url string, target interface{}) error {
+	r, err := myClient.Get(url)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+
+	return json.NewDecoder(r.Body).Decode(target)
+}
+
+func ConnectWithFence(gen3FuseConfig *Gen3FuseConfig) (err error) {
+	requestUrl := gen3FuseConfig.WTSBaseURL + gen3FuseConfig.WTSFenceConnectPath
+	fenceResponse := new(fenceConnectResponse)
+
+	getJson(requestUrl, fenceResponse)
+
+	if fenceResponse.Success != "connected with fence" {
+		FuseLog("Error connecting with fence via the workspace token service at " + requestUrl)
+		var fenceResponseStr string = fmt.Sprintf("%#v", fenceResponse)
+		FuseLog("The workspace token service came back with: " + fenceResponseStr)
+		return errors.New("Error connecting with fence via the workspace token service at " + requestUrl + ". Fence response: " + fenceResponseStr + 
+			".\n Are you sure the Workspace Token Service is configured correctly?")
+	}
+
+	return
+}
+
+func GetAccessToken(gen3FuseConfig *Gen3FuseConfig) (accessToken string) {
+	tokenLifetimeInSeconds := 3600
+	requestUrl := fmt.Sprintf(gen3FuseConfig.WTSBaseURL + gen3FuseConfig.WTSAccessTokenPath, tokenLifetimeInSeconds)
+
+	tokenResponse := new(tokenResponse)
+	getJson(requestUrl, tokenResponse)
```

Comment:
Patch `getJson` to return an `ok` flag that indicates that the endpoint returned HTTP 200 and JSON parse succeeded ...